collector/controller.go

```diff
@@ -28,13 +28,29 @@ func NewManager(cfg *rest.Config, options ctrl.Options) (ctrl.Manager, error) {
 	// and controller-runtime does not retry on missing CRDs.
 	// so we are going to wait on the CRDs existing before moving forward.
 	apiextensionsClient := apiextensionsclient.NewForConfigOrDie(cfg)
+	pipelineClient := pipelinev1client.NewForConfigOrDie(cfg)
 	if err := wait.PollImmediate(time.Second*5, time.Minute*5, func() (done bool, err error) {
 		_, err = apiextensionsClient.ApiextensionsV1().CustomResourceDefinitions().Get(context.TODO(), "pipelineruns.tekton.dev", metav1.GetOptions{})
 		if err != nil {
 			controllerLog.Error(err, "get of pipelinerun CRD failed")
 			return false, nil
 		}
 		controllerLog.Info("get of pipelinerun CRD returned successfully")
+		// in addition to the CRD check we've got in several controller-runtime based RHTAP controllers, metrics-exporter
+		// recently saw some intermittent issues even after this when setting up of watches or lists timed out as tekton
+		// was still ramping up, and controller runtime would exit out of initialization.  For example:
+		// "Failed to watch *v1.TaskRun: the server is currently unable to handle the request (get taskruns.tekton.dev)"
+		// "Failed to watch *v1.PipelineRun: the server is currently unable to handle the request (get pipelineruns.tekton.dev)"
+		// "failed to list *v1.TaskRun: the server was unable to return a response in the time allotted, but may still be processing the request (get taskruns.tekton.dev)"
+		// "Failed to watch *v1.TaskRun: failed to list *v1.TaskRun: the server was unable to return a response in the time allotted, but may still be processing the request (get taskruns.tekton.dev)"
+		//
+		// So we now try to see a list return successfully before we move on to controller-runtime initialization
+		_, err = pipelineClient.PipelineRuns("openshift-pipelines").List(context.TODO(), metav1.ListOptions{})
```

Comment:
nit: assuming the namespace "openshift-pipelines" does not have special meaning here, maybe use the more generic "default"?

Reply:
good question.

the rationale for using `openshit-pipelines` is that is where the exporter is installed, so we know it is there

I refrained from using `default` namespace because there are security issues with that namespace, and while it can not be deleted, there are actually security policies out there that will prevent pods launching in it.   I'll let you internet search if you are more curious

All that said, if I simply put an empty string there, it does a query across all pipelineruns in the cluster, which if successful is a better indication that the apiserver/controller are good to go, as we watch pipelineruns in all namespaces anyway.

I've verified it on a local cluster, will push momentarily.

Thanks.